project-base/storefront/pages/_app.tsx

```diff
@@ -35,26 +29,42 @@ process.on('uncaughtException', (error: Error, origin: unknown) =>
     }),
 );
 
+const UrqlWrapper = dynamic(() => import('components/Layout/UrqlWrapper').then((component) => component.UrqlWrapper), {
+    ssr: true,
+});
+
+const AppPageContent = dynamic(
```

Comment:
Hmm, I would say this breaks SSR (HTML returned from SSR). But I guess you have already checked it and it works, right? I am curious about how it works because otherwise I would though this would make app rendered only on client 🤔

Reply:
I analyzed the returned HTML, and saw these parts:

```html
 <li
                            tid="blocks_product_list_listeditem_532564"
                            class="relative flex select-none flex-col justify-between gap-3 border-greyLighter p-3 text-left lg:hover:z-above lg:hover:bg-white lg:hover:shadow-xl snap-center border-b-0 md:snap-start"
                        >
                            <a
                                class="flex h-full select-none flex-col gap-3 no-underline hover:no-underline"
                                draggable="false"
                                href="/canon-eh-22l"
                            >
                                <div class="relative flex h-56 items-center justify-center">
                                    <img
                                        alt="Product 7 image"
                                        draggable="false"
                                        loading="lazy"
                                        width="250"
                                        height="250"
                                        decoding="async"
                                        data-nimg="1"
                                        class="max-h-full object-contain"
                                        style="color: transparent"
                                        srcset="
                                            http://127.0.0.1:8000/content/images/product/canon-eh-22l_7.jpg?width=256 1x,
                                            http://127.0.0.1:8000/content/images/product/canon-eh-22l_7.jpg?width=640 2x
                                        "
                                        src="http://127.0.0.1:8000/content/images/product/canon-eh-22l_7.jpg?width=640"
                                    />
                                </div>
                                <div class="h-10 overflow-hidden text-lg font-bold leading-5 text-dark">
                                    Canon EH-22L
                                </div>
                                <div class="text-lg font-bold text-primary">€53.75</div>
                                <div class="flex flex-col gap-1 text-sm text-black">
                                    <div>In stock</div>
                                    <div class="text-sm text-inStock">
                                        This item is available immediately in 1 store
                                    </div>
                                </div>
                            </a>
```

or links in the menu

```html
                        <li class="group">
                            <a
                                class="relative m-0 flex items-center px-2 py-4 text-sm font-bold uppercase text-white no-underline hover:text-orangeLight hover:no-underline group-hover:text-orangeLight group-hover:no-underline vl:text-base"
                                href="/electronics"
                                >Gadgets</a
                            >
                        </li>
                        <li class="group">
                            <a
                                class="relative m-0 flex items-center px-2 py-4 text-sm font-bold uppercase text-white no-underline hover:text-orangeLight hover:no-underline group-hover:text-orangeLight group-hover:no-underline vl:text-base"
                                href="/books"
                                >Bookworm</a
                            >
                        </li>
                        <li class="group">
                            <a
                                class="relative m-0 flex items-center px-2 py-4 text-sm font-bold uppercase text-white no-underline hover:text-orangeLight hover:no-underline group-hover:text-orangeLight group-hover:no-underline vl:text-base"
                                href="/garden-tools"
                                >Growing</a
                            >
                        </li>
                        <li class="group">
                            <a
                                class="relative m-0 flex items-center px-2 py-4 text-sm font-bold uppercase text-white no-underline hover:text-orangeLight hover:no-underline group-hover:text-orangeLight group-hover:no-underline vl:text-base"
                                href="/food"
                                >Snack</a
                            >
                        </li>
```

or these blog links

```html
                                    <div class="mt-2 flex flex-col items-start gap-2">
                                        <a
                                            class="inline-flex rounded bg-primaryLight py-1 px-2 text-xs uppercase text-dark no-underline hover:text-dark hover:no-underline"
                                            href="http://127.0.0.1:8000/first-subsection-en"
                                            >First subsection en</a
                                        >
                                        <a
                                            class="block text-lg font-bold leading-5 text-white no-underline hover:text-white hover:underline"
                                            href="http://127.0.0.1:8000/blog-article-for-search-testing"
                                            >Blog article for search testing</a
                                        >
                                        <div class="leading-5">perex</div>
                                    </div>
                                </div>
                                <div
                                    class="hidden flex-1 flex-col text-white no-underline first:flex hover:text-white hover:no-underline lg:flex"
                                >
                                    <a
                                        class="block text-lg font-bold leading-5 text-white no-underline hover:text-white hover:underline"
                                        href="http://127.0.0.1:8000/blog-article-for-products-testing"
                                    >
                                        <img
                                            alt="Blog article for products testing"
                                            loading="lazy"
                                            width="700"
                                            height="220"
                                            decoding="async"
                                            data-nimg="1"
                                            class="mx-auto rounded"
                                            style="color: transparent"
                                            sizes="(max-width: 768px) 100vw, (max-width: 1024px) 50vw, 35vw"
                                            srcset="
                                                /images/optimized-noimage.webp?width=256   256w,
                                                /images/optimized-noimage.webp?width=384   384w,
                                                /images/optimized-noimage.webp?width=640   640w,
                                                /images/optimized-noimage.webp?width=750   750w,
                                                /images/optimized-noimage.webp?width=828   828w,
                                                /images/optimized-noimage.webp?width=1080 1080w,
                                                /images/optimized-noimage.webp?width=1200 1200w,
                                                /images/optimized-noimage.webp?width=1920 1920w,
                                                /images/optimized-noimage.webp?width=2048 2048w,
                                                /images/optimized-noimage.webp?width=3840 3840w
                                            "
                                            src="/images/optimized-noimage.webp?width=3840"
                                        />
                                    </a>
                                    <div class="mt-2 flex flex-col items-start gap-2">
                                        <a
                                            class="inline-flex rounded bg-primaryLight py-1 px-2 text-xs uppercase text-dark no-underline hover:text-dark hover:no-underline"
                                            href="http://127.0.0.1:8000/first-subsection-en"
                                            >First subsection en</a
                                        >
                                        <a
                                            class="block text-lg font-bold leading-5 text-white no-underline hover:text-white hover:underline"
                                            href="http://127.0.0.1:8000/blog-article-for-products-testing"
                                            >Blog article for products testing</a
                                        >
                                        <div class="leading-5">Blog article for products testing</div>
                                    </div>
                                </div>
                            </div>
```

I think it should be OK?

Reply:
Yes, sure!